src/clr-angular/data/datagrid/datagrid-column-toggle.ts

```diff
@@ -3,78 +3,94 @@
+import { ClrSmartPopoverToggleService } from '../../utils/smart-popover/providers/smart-popover-toggle.service';
 
 @Component({
   selector: 'clr-dg-column-toggle',
-  template: `
-        <button
-                #anchor
-                (click)="toggleUI()"
-                class="btn btn-sm btn-link column-toggle--action"
-                type="button">
-            <clr-icon shape="view-columns" [attr.title]="commonStrings.pickColumns"></clr-icon>
-        </button>
-        <div class="column-switch"
-             *clrPopoverOld="open; anchor: anchor; anchorPoint: anchorPoint; popoverPoint: popoverPoint">
-            <div class="switch-header">
-                <ng-container *ngIf="!title">Show Columns</ng-container>
-                <ng-content select="clr-dg-column-toggle-title"></ng-content>
-                <button
-                    class="btn btn-sm btn-link"
-                    (click)="toggleUI()"
-                    type="button">
-                    <clr-icon shape="close" [attr.title]="commonStrings.close"></clr-icon>
-                </button>
-            </div>
-            <ul class="switch-content list-unstyled">
-                <li *ngFor="let column of columns">
-                    <clr-checkbox-wrapper>
-                        <input clrCheckbox type="checkbox"
-                          [disabled]="column.lastVisibleColumn"
-                          [ngModel]="!column.hidden"
-                          (ngModelChange)="toggleColumn($event, column)">
-                        <label><ng-template [ngTemplateOutlet]="column.template"></ng-template></label>
-                    </clr-checkbox-wrapper>
-                </li>
-            </ul>
-            <div class="switch-footer" *ngIf="buttons.length > 0">
-                <ng-content select="clr-dg-column-toggle-button"></ng-content>
-            </div>
-            <div class="switch-footer" *ngIf="buttons.length === 0">
-                <div>
-                    <button
-                            class="btn btn-sm btn-link p6 text-uppercase"
-                            [disabled]="allColumnsVisible"
-                            (click)="selectAll()"
-                            type="button">Select All
-                    </button>
-                </div>
-            </div>
-        </div>
-    `,
-  host: { '[class.column-switch-wrapper]': 'true', '[class.active]': 'open' },
+  template: `    
+      <button
+              role="button"
+              type="button"
+              class="btn btn-sm btn-link column-toggle--action"
+              clrSmartAnchor
+              clrSmartOpenCloseButton
+              [attr.aria-owns]="popoverId">
+          <clr-icon shape="view-columns" [attr.title]="commonStrings.pickColumns"></clr-icon>
+      </button>
+      <div class="column-switch"
+           role="dialog"
+           [id]="popoverId"
+           clrFocusTrap
```

Comment:
As per our discussion, I am apprehensive of using `clrFocusTrap` here. Using a focus trap on modal dialog makes sense as we actually visually disable other parts of the application by putting an overlay on them so it creates a clear visual scope for users. For this popover dialog, there is no visual cue for users. Yet we have a focus trap on it, So users need to somehow figure out to close this popover. As such, I am not sure `clrFocusTrap` is a good candidate in this directive.